drivers/clock_control/nrf_power_clock.c

```diff
@@ -282,16 +322,15 @@ static void clkstarted_handle(struct device *dev)
 {
 	struct clock_control_async_data *async_data;
 	struct nrf_clock_control *data = dev->driver_data;
-	sys_snode_t *node = sys_slist_get(&data->list);
+
+	async_data = list_get(&data->list);
 
 	DBG(dev, "Clock started");
 	data->started = true;
 
-	while (node != NULL) {
-		async_data = CONTAINER_OF(node,
-				struct clock_control_async_data, node);
+	while (async_data != NULL) {
```

Comment:
What about:
```
	for (;;) {
		struct clock_control_async_data *async_data;
		
		async_data = list_get(&data->list);
		if (!async_data) {
			break;
		}

		async_data->cb(dev, async_data->user_data);
	}
```
The generated code should be simpler then, as `list_get` is called only once.
Or in a shorter version:
```
	while ((async_data = list_get(&data->list) != NULL) {
		async_data->cb(dev, async_data->user_data);
	}
```

Reply:
done